h/browser/chrome/lib/install.js

```diff
@@ -14,7 +14,16 @@ var browserExtension = new HypothesisChromeExtension({
 });
 
 browserExtension.listen(window);
+
+// Respond to messages sent by the JavaScript from https://hpt.is.
+// This is how it knows whether the user has this Chrome extension installed.
 chrome.runtime.onInstalled.addListener(onInstalled);
+
+chrome.runtime.onMessageExternal.addListener(
+  function (request, sender, sendResponse) {
```

Comment:
We're not doing any kind of validation here that the message is actually a request from the bouncer. That is the only external thing that is allowed to connect right now, but I think it would make sense to check the message type (if the request is an object and has a particular 'type' property) and send a suitable type in response.